src/dev/flang/util/Errors.java

```diff
@@ -349,7 +353,19 @@ public static void fatal(String s, String detail)
   {
     error(s, detail);
     System.err.println("*** fatal errors encountered, stopping.");
-    System.exit(1);
+    exit(1);
+  }
+
+
+  /**
+   * Call System.exit in case preventExit is false.
+   */
+  private static void exit(int status)
+  {
+    if (!preventExit)
+      {
+        System.exit(status);
+      }
```

Comment:
Just returning from a `Error.fatal` is IMHO not a good idea, the code that calls `Error.fatal` can safely assume this not to return. If this returns anyway, we might run into more errors or endless loops, endless recursions, etc. 

The only alternative I see is to declare a new class FatalError that extends Error and throw this. Then, `Tool.run` or the corresponding routine in the language server can catch `FatalError` and handle it. 

Maybe we could even get rid of the `preventExit` flag and always terminate via `FatalError`, maybe `FatalError` could contain the `status` code as well, so `Tool.run` could then catch it and `System.exit` with this status. 

Futhermore, it might be good to check for repeated calls to `Errors.exit` and perform a `System.exit` in this case (just in case some code catches `Error` and runs into a loop peroducing fatal errors.